### PR TITLE
Auth changes for skills

### PR DESCRIPTION
**Pull request by axelsrz**
Solves #342

### CODE REVIEW

File: libraries/botframework-connector/botframework/connector/auth/jwt_token_validation.py [hunk 1]
```diff
@@ -61,13 +68,21 @@ class JwtTokenValidation:
...
     ) -> ClaimsIdentity:
         if not auth_header:
             raise ValueError("argument auth_header is null")
 
-        using_emulator = EmulatorValidation.is_token_from_emulator(auth_header)
+        if SkillValidation.is_skill_token(auth_header):
+            return await SkillValidation.authenticate_channel_token(
+                auth_header,
+                credentials,
+                channel_service,  # TODO
```
Comment (carlosscastro) [inline libraries/botframework-connector/botframework/connector/auth/jwt_token_validation.py]: what;s this todo?

Comment (carlosscastro) [reply libraries/botframework-connector/botframework/connector/auth/jwt_token_validation.py]: i.e. what is pending here?

**Comment by carlosscastro** [comment]
Changes look good overall. What is the plan to get these in? Include all skill work and merge in one PR or merge this in and then the rest of skills?

Ideally I'd love to merge this in once we have 1) functional tests for skills and 2) functional auth tests to make sure that the regular path authentication has not been affected.

File: libraries/botframework-connector/botframework/connector/auth/jwt_token_validation.py (hunk 1 above)
Comment (axelsrz) [reply libraries/botframework-connector/botframework/connector/auth/jwt_token_validation.py]: I should remove that comment, I was considering porting the ChannelProvider interface but I remembered I had that functionality in a helper class (JS do it that way)

**Comment by tracyboehrer** [comment]
> Changes look good overall. What is the plan to get these in? Include all skill work and merge in one PR or merge this in and then the rest of skills?
> 
> Ideally I'd love to merge this in once we have 1) functional tests for skills and 2) functional auth tests to make sure that the regular path authentication has not been affected.

In the Teams case, Josh is pushing to another branch.  The plan is to then merge that into master when done.  Would we not want to do the same here so we don't end up with something incomplete if we we don't finish?

File: libraries/botframework-connector/botframework/connector/auth/jwt_token_extractor.py [hunk 2]
```diff
@@ -64,27 +73,46 @@ def _has_allowed_issuer(self, jwt_token: str) -> bool:
         if issuer in self.validation_parameters.issuer:
             return True
 
-        return issuer is self.validation_parameters.issuer
+        return issuer == self.validation_parameters.issuer
```
Comment (stevengum) [inline libraries/botframework-connector/botframework/connector/auth/jwt_token_extractor.py]: Good fix!

File: libraries/botframework-connector/botframework/connector/auth/jwt_token_extractor.py [hunk 3]
```diff
@@ -33,17 +34,23 @@ def get_open_id_metadata(metadata_url: str):
...
+                parts[0], parts[1], channel_id, required_endorsements
+            )
         return None
 
     async def get_identity(
-        self, schema: str, parameter: str, channel_id
+        self,
+        schema: str,
+        parameter: str,
+        channel_id,
```
Comment (stevengum) [inline libraries/botframework-connector/botframework/connector/auth/jwt_token_extractor.py]: `channel_id: str,` ?

**Pull request merged**